test/pretty-printer/ppTest.ml

```diff
@@ -71,6 +75,7 @@ let arbitrary_ast =
           shrink_binary (Expr.letc idr) bound body
       | Expr.Letbox { idm; boxed; body } ->
           shrink_binary (Expr.letbox idm) boxed body
+      | Expr.Match _ -> empty
```

Comment:
Here, we also need to shrink the components of the `match`-expression.

Reply:
I fixed it, but I don't know is my solution correct.

Reply:
Here you need to shrink all three subexpressions of the `match`-expression.
So you could define a special version of a shrinker for ternary constructors like so (put it right after `shrink_binary`):
```
    let shrink_ternary cons arg1 arg2 arg3 =
      of_list [ arg1; arg2; arg3 ]
      <+> (shrink_ast arg1 >|= fun arg1' -> cons arg1' arg2 arg3)
      <+> (shrink_ast arg2 >|= fun arg2' -> cons arg1 arg2' arg3)
      <+> (shrink_ast arg3 >|= fun arg3' -> cons arg1 arg2 arg3')
in
...
```
and then use it like this

```
shrink_ternary (fun m z s -> Expr.match_with m z pred s) matched zbranch sbranch
```